### ISSUE TITLE
ZeroDivisionError in plot-cn

### ISSUE THREAD

**Issue by DaGaMs** [raphael-group/hatchet#64]
In one of my samples, I get this error when trying to plot:

```
# Plotting reduced-clone profiles in ./intratumor-profilesreduced.pdf
m/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py:704: UserWarning: Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.
  ax.set_xlim(0, max_dependent_coord * 1.05)
# Plotting reduced mixtures in ./intratumor-mixtures.pdf
Traceback (most recent call last):
  File "/opt/gridware/depots/1a8f5697/el7/pkg/apps/python3/3.8.1/gcc-4.8.5/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/gridware/depots/1a8f5697/el7/pkg/apps/python3/3.8.1/gcc-4.8.5/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/hatchet/__main__.py", line 69, in <module>
    sys.exit(main())
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/hatchet/__main__.py", line 63, in main
    globals()[command](args)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/hatchet/utils/plot_cn.py", line 159, in main
    single(tumors[list(tumors)[0]], clones[list(clones)[0]], props[list(props)[0]], base[list(base)[0]], args)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/hatchet/utils/plot_cn.py", line 212, in single
    gridmixtures(tumor, base, clones, props, args, out)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/hatchet/utils/plot_cn.py", line 536, in gridmixtures
    g = sns.clustermap(**para)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/_decorators.py", line 46, in inner_f
    return f(**kwargs)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 1408, in clustermap
    return plotter.plot(metric=metric, method=method,
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 1221, in plot
    self.plot_dendrograms(row_cluster, col_cluster, metric, method,
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 1066, in plot_dendrograms
    self.dendrogram_row = dendrogram(
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/_decorators.py", line 46, in inner_f
    return f(**kwargs)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 774, in dendrogram
    plotter = _DendrogramPlotter(data, linkage=linkage, axis=axis,
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 584, in __init__
    self.linkage = self.calculated_linkage
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 651, in calculated_linkage
    return self._calculate_linkage_scipy()
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/seaborn/matrix.py", line 619, in _calculate_linkage_scipy
    linkage = hierarchy.linkage(self.array, method=self.method,
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/scipy/cluster/hierarchy.py", line 1060, in linkage
    y = distance.pdist(y, metric)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/scipy/spatial/distance.py", line 2072, in pdist
    dm[k] = metric(X[i], X[j], **kwargs)
  File "/opt/apps/pkg/apps/hatchet/0.3.1/gcc-4.8.5+python3-3.8.1+gurobi-9.1.2+samtools-1.9+bcftools-1.9/lib/python3.8/site-packages/hatchet/utils/plot_cn.py", line 939, in similaritysample
    return float(sum((bothamp(u[i], v[i]) or bothdel(u[i], v[i])) and u[i] != 0 and v[i] != 0 for i in range(len(u)))) / float(sum(u[i] != 0 or v[i] != 0 for i in range(len(u))))
ZeroDivisionError: float division by zero
```

The input for that HATCHet run were two tumour samples vs one control. It seems the tumour samples had low purity and/or no CNVs, ie this might just be a bad sample. Still, I guess one should guard against crashes even if the input is poor quality. 

Let me know if you need the `.ucn` files for debugging.

**Comment by simozacca** [comment]
Thank you for your feedback!

The error should have been fixed in 81fa8ae42366a9e126da5c2f5095ebef5350673b ; if you could please send us your `best.seg.ucn` and `best.bbc.ucn` files in `results` we would be very happy to test if this change fixes your issue before release.

**Comment by simozacca** [comment]
We assume that the error has been fixed. Please feel free to re-open the issue otherwise or in case of other issues.
